Userland/Libraries/LibWeb/PixelUnits.cpp

```diff
@@ -0,0 +1,157 @@
+{
+    m_value = value * fixed_point_denominator;
+}
+
+CSSPixels::CSSPixels(double value)
+{
+    m_value = value * fixed_point_denominator;
+}
+
+float CSSPixels::to_float() const
+{
+    return static_cast<float>(m_value) / fixed_point_denominator;
+}
+
+double CSSPixels::to_double() const
+{
+    return static_cast<double>(m_value) / fixed_point_denominator;
+}
+
+int CSSPixels::to_int() const
+{
+    return m_value / fixed_point_denominator;
+}
+
+bool CSSPixels::operator==(CSSPixels const& other) const
+{
+    return raw_value() == other.raw_value();
+}
+
+CSSPixels& CSSPixels::operator++()
+{
+    m_value += fixed_point_denominator;
+    return *this;
+}
+CSSPixels& CSSPixels::operator--()
+{
+    m_value -= fixed_point_denominator;
+    return *this;
+}
+
+int CSSPixels::operator<=>(CSSPixels const& other) const
+{
+    return this->raw_value() > other.raw_value() ? 1 : this->raw_value() < other.raw_value() ? -1
+                                                                                             : 0;
+}
+
+int saturated_addition(int a, int b)
+{
+    i32 sum = a + b;
+    i32 overflow = (b > 0 && a > NumericLimits<int>::max() - b);
+    i32 underflow = (b < 0 && a < NumericLimits<int>::min() - b);
+    return overflow ? NumericLimits<int>::max() : (underflow ? NumericLimits<int>::min() : sum);
+}
+
+CSSPixels CSSPixels::operator+() const
+{
+    CSSPixels result;
+    result.set_raw_value(+this->raw_value());
+    return result;
+}
+
+CSSPixels CSSPixels::operator-() const
+{
+    CSSPixels result;
+    result.set_raw_value(-this->raw_value());
+    return result;
+}
+
+CSSPixels CSSPixels::operator+(CSSPixels const& other) const
+{
+    CSSPixels result;
+    result.set_raw_value(saturated_addition(this->raw_value(), other.raw_value()));
+    return result;
+}
+
+CSSPixels CSSPixels::operator-(CSSPixels const& other) const
+{
+    CSSPixels result;
+    result.set_raw_value(this->raw_value() - other.raw_value());
+    return result;
+}
+
+CSSPixels CSSPixels::operator*(CSSPixels const& other) const
+{
+    CSSPixels result;
+    result.set_raw_value((this->raw_value() * other.raw_value()) >> layout_unit_fractional_bits);
+    return result;
+}
+
+CSSPixels CSSPixels::operator/(CSSPixels const& other) const
+{
+    CSSPixels result;
+    result.set_raw_value((this->raw_value() << layout_unit_fractional_bits) / other.raw_value());
+    return result;
+}
+
+CSSPixels& CSSPixels::operator+=(CSSPixels const& other)
+{
+    *this = *this + other;
+    return *this;
+}
+
+CSSPixels& CSSPixels::operator-=(CSSPixels const& other)
+{
+    *this = *this - other;
+    return *this;
+}
+
+CSSPixels& CSSPixels::operator*=(CSSPixels const& other)
+{
+    *this = *this * other;
+    return *this;
+}
+
+CSSPixels& CSSPixels::operator/=(CSSPixels const& other)
+{
+    *this = *this / other;
+    return *this;
+}
+
+CSSPixels CSSPixels::abs() const
+{
+    CSSPixels result;
+    result.set_raw_value(::abs(m_value));
+    return result;
+}
```

Comment:
You could add overloads for common functions like this, so things work like regular floats.
```c++
CSSPixels abs(CSSPixels value) {
   return value.abs();
}
```

Reply:
done